simpeg/directives/_regularization.py

```diff
@@ -0,0 +1,427 @@
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+from dataclasses import dataclass
+
+from .directives import InversionDirective, UpdatePreconditioner, BetaSchedule
+from simpeg.regularization import Sparse, BaseSparse, SmoothnessFirstOrder
+from simpeg.utils import validate_integer, validate_float
+
+
+@dataclass
+class IRLSMetrics:
+    """
+    Data class to store metrics used by the IRLS algorithm.
+
+    Parameters
+    ----------
+
+    input_norms: List of norms temporarily stored during the initialization.
+    irls_iteration_count: Number of IRLS iterations.
+    start_irls_iter: Iteration number when the IRLS process started.
+    f_old: Previous value of the regularization function.
```

Comment:
Update docstring to numpy docstyle.
```suggestion
    input_norms : list of floats
        List of norms temporarily stored during the initialization.
    irls_iteration_count : int
        Number of IRLS iterations.
    start_irls_iter : int or None
        Iteration number when the IRLS process started.
    f_old : float
        Previous value of the regularization function.
```